frontends/common/constantFolding.cpp

```diff
@@ -50,6 +50,12 @@ class CloneConstants : public Transform {
 
 const IR::Expression* DoConstantFolding::getConstant(const IR::Expression* expr) const {
     CHECK_NULL(expr);
+    if (auto pathexpr = expr->to<IR::PathExpression>()) {
```

Comment:
The reason for this is that the processing for the cast has already been implemented, but it did not work because the expression was PathExpression and its search returned Declaration_Variable instead of Declaration_Constatn

Reply:
I do not know the constantFolding pass very well but this change does not appear correct to me. For example, you do not even check if you are returning a constant. And it breaks a lot of existing tests. Shouldn't you handle this in the unrolling pass? Or do you think there is a bug in constantFolding?

Reply:
It seems to me that there was not enough implementation for the cases with PathExpression, but they are caused by our change of bool to cast. Probably back to the original version